bittensor/_metagraph/metagraph_impl.py

```diff
@@ -247,7 +247,7 @@ def modalities( self ) -> List[str]:
         """
         if self.n.item() == 0:
             return []
-        return [ neuron.modality if neuron != None else '' for neuron in self.endpoint_objs ]
+        return [ neuron.modality if neuron.modality != None else '' for neuron in self.endpoint_objs ]
```

Comment:
What is your thinking here? this will fail if the endpoint is Null?

Reply:
The custom endpoint.__eq__ breaks the comparison in neuron != None, therefore breaking metagraph.hotkeys/coldkeys/modalities